pandas/tests/series/test_constructors.py

```diff
@@ -625,6 +625,26 @@ def test_constructor_dict(self):
         expected.iloc[1] = 1
         assert_series_equal(result, expected)
 
+    def test_constructor_dict_nan_key(self):
+        # GH 18480
+        d = {1: 'a', 2: 'b', np.nan: 'c'}
+        result = Series(d).sort_index()
+        expected = Series(['a', 'b', 'c'], index=[1, 2, np.nan])
+        assert_series_equal(result, expected)
+
+        # Different NaNs:
+        d = {1: 'a', 2: 'b', float('nan'): 'c', float('nan'): 'd'}
+        result = Series(d).sort_values()
+        expected = Series(['a', 'b', 'c', 'd'], index=[1, 2, np.nan, np.nan])
```

Comment:
Is the order of the expected result guaranteed like this? (I mean: I though we sort the index, but not the values?)

Reply:
That's why I'm ``sort_values()``ing just above ;-)

Reply:
Ah, didn't see that :-)